### ISSUE TITLE
Validate color--redshift discreteness 

### ISSUE THREAD

**Issue by aphearin** [LSSTDESC/lsstdesc-diffsky#33]
As reported by @sidneymau the small-volume test catalog diffsky, `diffsky_v0.1_p3765_vsmall`, presents color--redshift discreteness effects:
![image](https://github.com/LSSTDESC/lsstdesc-diffsky/assets/6951595/434b7b39-925a-47c0-8da1-f8c0c29d4d6b)

This behavior has been seen before in previous iterations of Diffsky models that populated a single healpixel, and we observed that the cause was cosmic variance since the effect entirely vanished when populating a cosmoDC2-sized sky area (see plot below). But this should be re-confirmed with future larger-volume productions of the diffsky mocks.

![ri_obs_single_heapixel_vs_cosmodc2](https://github.com/LSSTDESC/lsstdesc-diffsky/assets/6951595/d2ab31bc-a0a2-4642-bc1a-0a8bfcb96bcc)

**Comment by sidneymau** [comment]
Following up here. I ran this same code on the `roman_rubin_2023_v1.1.1` catalog. I am still seeing some striations in the color vs redshift distributions. Here are plots for both g-r and r-i color vs redshift.
![image](https://github.com/LSSTDESC/lsstdesc-diffsky/assets/26752606/890b798f-8ecc-4169-9cce-d0f1d73f8200)
![image](https://github.com/LSSTDESC/lsstdesc-diffsky/assets/26752606/19ed345d-64d7-423c-8799-cfb66522f076)

**Comment by aphearin** [comment]
Thanks for following up on this @sidneymau. Hmmm, yes definitely still visible. Although also appears to be quite a bit less pronounced than the single-healpixel plot, don’t you think?

This could be due to the redshift table used in the calculation of the approximate magnitudes, or it could be something else. A very pure test of that would be to remake this plot using the exact photometry (remember the approximate photometry is what we write to disk and store in the columns).

**Comment by sidneymau** [comment]
Yes, definitely less pronounced than from the single-healpixel plot. Good point about the approximate photometry... especially given the interpolation on the grid of redshift. If the redshift grid is logarithmically spaced that would also explain how the spacing between striations seems to lengthen w.r.t. redshift.

I can try and make a version with computing the exact photometry some time in the next few days.

**Comment by aphearin** [comment]
So I still think the test using the exact photometry is worth doing, but after thinking about this a little more it still seems like the most plausible explanation is cosmic variance, since the effect reduces so much when going from one healpixel to many. It's worth pointing out that these vertical striations are also seen directly in observational data of small galaxy samples. See for example the left panel of Figure 5 of this paper that posted to the arXiv today: [https://arxiv.org/abs/2311.10469](https://arxiv.org/abs/2311.10469)
